experimental/PartitionedPermutations/src/EnumeratePartitionedPermutations.jl

```diff
@@ -0,0 +1,68 @@
+"""
+    _enumerate_all_partitions(n::Int)
+
+Return and efficiently calculate vector of all partitions of length `n`.
+This function is a helper function for `_enumerate_partitioned_perm`.
+
+# Examples
+```jldoctest
+julia> length(_enumerate_all_partitions(6))
+203
+```
+"""
+function _enumerate_all_partitions(n::Int)
+
+    partitions = [[1]]
+    index_to_cut = 0
+    count = false
+    
+    for i in partitions
+        blocks = maximum(i)
+        length(i) == n && continue
+        count = (length(i) == n - 1)
+        for ii in 1:(blocks+1)
+            push!(partitions, vcat(i, ii))
+            if count
+                index_to_cut += 1
+            end
+        end
+    end
+    return last(partitions, index_to_cut == 0 ? 1 : index_to_cut)
+end
+
+"""
+    enumerate_partitioned_perm(n::Int)
+
+Return and calculate all `PartitionedPermutation` objects of length `n`
+
+# Examples
+```jldoctest
+julia> length(enumerate_partitioned_perm(6))
+4051
+```
+"""
+function enumerate_partitioned_perm(n::Int)
```

Comment:
```suggestion
function enumerate_partitioned_permutations(n::Int)
```
For consistency with the other exported `*permutation` function.